File: 7-kyu/which-triangle-is-that-/javascript/solution.js
const typeOfTriangle = (sideA, sideB, sideC) => {
  if (typeof sideA !== "number" || typeof sideB !== "number" || typeof sideC !== "number") {
    return "Not a valid triangle";
  }
  const sides = [sideA, sideB, sideC].sort((a, b) => a - b);
  if (sides[0] + sides[1] > sides[2]) {
    if (sideA === sideB && sideB === sideC) {
      return "Equilateral";
    } else if (sideA === sideB || sideB === sideC || sideA === sideC) {
      return "Isosceles";
    }
    return "Scalene";
  }
  return "Not a valid triangle";
}

File: 7-kyu/which-triangle-is-that-/python/solution.py
def type_of_triangle(a, b, c): 
    if any(not isinstance(elem, int) for elem in [a, b, c]):
        return 'Not a valid triangle'
    shortest, mid, longest = sorted([a, b, c])
    if shortest + mid > longest:
        if a == b and b == c:
            return 'Equilateral'
        elif a == b or b == c or a == c:
            return 'Isosceles'
        return 'Scalene'
    return 'Not a valid triangle'
